File: OSCommon/org/optimizationservices/oscommon/datastructure/osgeneral/SparseVector.cs
using System;

namespace org.optimizationservices.oscommon.datastructure{
	/// <summary>
	/// The <c>SparseVector</c> class holds a two array 
	/// data structure for sparse representation of vector. 
	/// @author Jun Ma	
	/// @version 1.0, 09/01/2005
	/// @since OS 1.0
	/// @copyright (c) 2005
	/// </summary>
	public class SparseVector{

		/// <summary>
		/// indexes holds an integer array of indexes, which corresponding values are nonzero.
		/// </summary>
		public int[] indexes = null;

		/// <summary>
		/// values holds a double array of nonzero values.
		/// </summary>
		public double[] values = null;

		/// <summary>
		/// Default constructor. 
		/// </summary>
		public SparseVector(){
		}//constructor

	}//class SparseVector
}//namespace

File: OSCommon/src/org/optimizationservices/oscommon/datastructure/osgeneral/SparseVector.java
/**
 * @(#)SparseVector 1.0 03/14/2004
 *
 * Copyright (c) 2004
 */
package org.optimizationservices.oscommon.datastructure.osgeneral;

/**
*
* <P>The <code>SparseVector</code> class holds a two array 
* data structure for sparse representation of vector. 
*  
* </p>
* @author Robert Fourer, Jun Ma, Kipp Martin
* @version 1.0, 03/14/2004
* @since OS 1.0
*/

public class SparseVector {

	
	/**
	 * indexes holds an integer array of indexes, which corresponding values are nonzero.
	 */
	public int[] indexes = null;

	/**
	 * values holds a double array of nonzero values.
	 */
	public double[] values = null;

	/**
	 *
	 * Default constructor. 
	 */
	public SparseVector(){
	}//constructor
	
	/**
	 * main for test purposes.
	 *
	 * </p>
	 *
	 * @param argv command line arguments.
	 */
	public static void main(String[] args) {
	}//main

}//SparseVector
